src/agent0/ethpy/hyperdrive/interface/read_interface.py

```diff
@@ -209,6 +209,16 @@ def __init__(
         self._current_pool_state = None
         self.last_state_block_number = -1
 
+        # Best effort to find initialize event and set deploy block
+        self.deploy_block: None | int = None
+        initialize_event = self.get_initialize_events("earliest")
```

Comment:
Are we sure we want this every time the interface is initialized? If I understand correctly this will pull all logs from the earliest available block. That has to be really time consuming, right? I think we should populate this when requested (`interface.get_deploy_block()`) and make sure to make that call in the invariance checks.